src/forms/create_ticket.ts

```diff
@@ -20,3 +31,206 @@ export async function newCreateTicketForm(call: AppCall): Promise<AppForm> {
     };
     return form;
 }
+
+// FormFields retrieves viewable modal app fields
+class FormFields extends BaseFormFields {
+    zdTicketForms: zdFormFieldOption[]
+    postMessage: string
+    constructor(call: AppCall, zdClient: ZDClient, mmClient: Client4) {
+        super(call, zdClient, mmClient);
+        this.postMessage = call.context.post.message;
+        this.zdTicketForms = [];
+    }
+
+    // getFields returns a list of viewable app fields mapped from Zendesk form fields
+    async getCreateTicketFields(): Promise<AppField[]> {
+        await this.setState();
+        await this.buildFields();
+        return this.builder.getFields();
+    }
+
+    // buildFields adds fields to list of viewable proxy app fields
+    async buildFields(): Promise<void> {
+        // show the form selector when intially opening the modal
+        this.addFormsSelectField();
+
+        // only show form select field until user selects a form select value
+        if (!this.builder.currentFieldValuesAreDefined()) {
+            return;
+        }
+
+        // add fields that are dependant on form selector pulldown being defined
+        await this.addFormSelectDependentFields();
+    }
+
+    // setState sets state for the current Zendesk Form
+    async setState(): Promise<void> {
```

Comment:
I would avoid using this naming, usually we are using `setState` to initialize the fields of React's `this.state`, which we do not use here (maybe we should?)